[Attack Discovery][Scheduling] Add cases action support

e40pud opened the pull request:
## Summary

With these changes we add Case Action support to Attack Discovery Schedule rule types.

Attack discovery alerts act differently from SIEM alerts and include the reference to list of SIEM alerts that led to the attack - described within the attack discovery alert document. Thus, we would like to attach referenced SIEM alerts instead of the attack alert document itself to the created Case. Also, as part of the Case creation we would like to be able to add a comment generated by LLM that describes steps and nuance of the discovery.

## NOTES

The attack discovery scheduling feature is hidden behind the feature flag (in `kibana.dev.yml`):

```
feature_flags.overrides:
  securitySolution.attackDiscoveryAlertsEnabled: true
```


<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

e40pud committed:
[Attack Discovery][Scheduling] Add cases action support

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

e40pud committed:
Fix CI

e40pud committed:
Fix CI

e40pud committed:
Fix CI

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/305906)
* Commit: a71bcab3c57a55a8b172de677813b89a89d5b879

### Failed CI Steps
* [Jest Integration Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/305906#0197411c-9a25-4117-8926-4c20091caf53)
* [Jest Integration Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/305906#01974141-b84f-4add-b486-784d5f54217f)
* [FTR Configs #70](https://buildkite.com/elastic/kibana-pull-request/builds/305906#0197411c-9a78-4bd4-b401-2395d02d664a)
* [FTR Configs #122](https://buildkite.com/elastic/kibana-pull-request/builds/305906#0197411c-9aa2-499a-8628-d3663360e9f7)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/305906#0197411c-9a25-4117-8926-4c20091caf53) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/305906/jobs/0197411c-9a25-4117-8926-4c20091caf53/artifacts/01974141-9e15-402f-a37a-64dbb5a6f99a) Jest Integration Tests #<span></span>8 / Connector type config checks detect connector type changes for: .cases
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/305906#01974141-b84f-4add-b486-784d5f54217f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/305906/jobs/01974141-b84f-4add-b486-784d5f54217f/artifacts/0197414c-439c-4e51-a657-3e17c2d23119) Jest Integration Tests #<span></span>8 / Connector type config checks detect connector type changes for: .cases

### History
* :broken_heart: [Build #305891](https://buildkite.com/elastic/kibana-pull-request/builds/305891) failed a9db23878ae69e14b53b2cfcf4d72985fc61d16c
* :broken_heart: [Build #305848](https://buildkite.com/elastic/kibana-pull-request/builds/305848) failed 2fc163627d92a58b8d43b47c3eadf3e66d680c2b
* :broken_heart: [Build #305824](https://buildkite.com/elastic/kibana-pull-request/builds/305824) failed 61be697aaf6afd7e84468f1944fbe72e1603087e

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305906","number":305906,"commit":"a71bcab3c57a55a8b172de677813b89a89d5b879"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305891","number":305891,"commit":"a9db23878ae69e14b53b2cfcf4d72985fc61d16c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305848","number":305848,"commit":"2fc163627d92a58b8d43b47c3eadf3e66d680c2b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305824","number":305824,"commit":"61be697aaf6afd7e84468f1944fbe72e1603087e"}],"number":305906}
buildkite-pr-comment-->

e40pud committed:
Fix CI

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307299)
* Commit: d5e51e0c5f7224da63fa56e5e677e67bd6c3ca17
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-d5e51e0c5f72`

### History
* :green_heart: [Build #306959](https://buildkite.com/elastic/kibana-pull-request/builds/306959) succeeded faafa04d4e10129a615a8bc9a55a2d65da3ad16f
* :broken_heart: [Build #305906](https://buildkite.com/elastic/kibana-pull-request/builds/305906) failed a71bcab3c57a55a8b172de677813b89a89d5b879
* :broken_heart: [Build #305891](https://buildkite.com/elastic/kibana-pull-request/builds/305891) failed a9db23878ae69e14b53b2cfcf4d72985fc61d16c
* :broken_heart: [Build #305848](https://buildkite.com/elastic/kibana-pull-request/builds/305848) failed 2fc163627d92a58b8d43b47c3eadf3e66d680c2b
* :broken_heart: [Build #305824](https://buildkite.com/elastic/kibana-pull-request/builds/305824) failed 61be697aaf6afd7e84468f1944fbe72e1603087e

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307299","number":307299,"commit":"d5e51e0c5f7224da63fa56e5e677e67bd6c3ca17"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306959","number":306959,"commit":"faafa04d4e10129a615a8bc9a55a2d65da3ad16f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305906","number":305906,"commit":"a71bcab3c57a55a8b172de677813b89a89d5b879"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305891","number":305891,"commit":"a9db23878ae69e14b53b2cfcf4d72985fc61d16c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305848","number":305848,"commit":"2fc163627d92a58b8d43b47c3eadf3e66d680c2b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305824","number":305824,"commit":"61be697aaf6afd7e84468f1944fbe72e1603087e"}],"number":307299}
buildkite-pr-comment-->

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307597)
* Commit: 40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-40c6fa2fcd75`

### History
* :green_heart: [Build #307299](https://buildkite.com/elastic/kibana-pull-request/builds/307299) succeeded d5e51e0c5f7224da63fa56e5e677e67bd6c3ca17
* :green_heart: [Build #306959](https://buildkite.com/elastic/kibana-pull-request/builds/306959) succeeded faafa04d4e10129a615a8bc9a55a2d65da3ad16f
* :broken_heart: [Build #305906](https://buildkite.com/elastic/kibana-pull-request/builds/305906) failed a71bcab3c57a55a8b172de677813b89a89d5b879
* :broken_heart: [Build #305891](https://buildkite.com/elastic/kibana-pull-request/builds/305891) failed a9db23878ae69e14b53b2cfcf4d72985fc61d16c
* :broken_heart: [Build #305848](https://buildkite.com/elastic/kibana-pull-request/builds/305848) failed 2fc163627d92a58b8d43b47c3eadf3e66d680c2b
* :broken_heart: [Build #305824](https://buildkite.com/elastic/kibana-pull-request/builds/305824) failed 61be697aaf6afd7e84468f1944fbe72e1603087e

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307597","number":307597,"commit":"40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307299","number":307299,"commit":"d5e51e0c5f7224da63fa56e5e677e67bd6c3ca17"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306959","number":306959,"commit":"faafa04d4e10129a615a8bc9a55a2d65da3ad16f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305906","number":305906,"commit":"a71bcab3c57a55a8b172de677813b89a89d5b879"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305891","number":305891,"commit":"a9db23878ae69e14b53b2cfcf4d72985fc61d16c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305848","number":305848,"commit":"2fc163627d92a58b8d43b47c3eadf3e66d680c2b"}],"number":307597}
buildkite-pr-comment-->

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307729)
* Commit: 239b67b1ce45e1e73813d1870e29fca3eecbb02b
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-239b67b1ce45`

### History
* :broken_heart: [Build #307656](https://buildkite.com/elastic/kibana-pull-request/builds/307656) failed 3304a47fd1d678c258f00d78b9243f52f587dde5
* :green_heart: [Build #307627](https://buildkite.com/elastic/kibana-pull-request/builds/307627) succeeded 40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5
* :broken_heart: [Build #307597](https://buildkite.com/elastic/kibana-pull-request/builds/307597) failed 40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5
* :green_heart: [Build #307299](https://buildkite.com/elastic/kibana-pull-request/builds/307299) succeeded d5e51e0c5f7224da63fa56e5e677e67bd6c3ca17
* :green_heart: [Build #306959](https://buildkite.com/elastic/kibana-pull-request/builds/306959) succeeded faafa04d4e10129a615a8bc9a55a2d65da3ad16f

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307729","number":307729,"commit":"239b67b1ce45e1e73813d1870e29fca3eecbb02b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307656","number":307656,"commit":"3304a47fd1d678c258f00d78b9243f52f587dde5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307627","number":307627,"commit":"40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307597","number":307597,"commit":"40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307299","number":307299,"commit":"d5e51e0c5f7224da63fa56e5e677e67bd6c3ca17"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306959","number":306959,"commit":"faafa04d4e10129a615a8bc9a55a2d65da3ad16f"}],"number":307729}
buildkite-pr-comment-->

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308212)
* Commit: 8ce3913429ec94767b6f8b6f943b2cb7b4f7eb1c
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-8ce3913429ec`

### History
* :green_heart: [Build #308134](https://buildkite.com/elastic/kibana-pull-request/builds/308134) succeeded 85f549933521750c61522d9026ec727a9c0b3de1
* :green_heart: [Build #307729](https://buildkite.com/elastic/kibana-pull-request/builds/307729) succeeded 239b67b1ce45e1e73813d1870e29fca3eecbb02b
* :broken_heart: [Build #307656](https://buildkite.com/elastic/kibana-pull-request/builds/307656) failed 3304a47fd1d678c258f00d78b9243f52f587dde5
* :green_heart: [Build #307627](https://buildkite.com/elastic/kibana-pull-request/builds/307627) succeeded 40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5
* :broken_heart: [Build #307597](https://buildkite.com/elastic/kibana-pull-request/builds/307597) failed 40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308212","number":308212,"commit":"8ce3913429ec94767b6f8b6f943b2cb7b4f7eb1c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308134","number":308134,"commit":"85f549933521750c61522d9026ec727a9c0b3de1"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307729","number":307729,"commit":"239b67b1ce45e1e73813d1870e29fca3eecbb02b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307656","number":307656,"commit":"3304a47fd1d678c258f00d78b9243f52f587dde5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307627","number":307627,"commit":"40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307597","number":307597,"commit":"40c6fa2fcd75e2a0d1221a8e19ec8437fce0e6e5"}],"number":308212}
buildkite-pr-comment-->

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Review feedback: Make cases connector executor rule type agnostic

e40pud committed:
Remove obsolete code and update tests

e40pud committed:
Revert redundant changes

e40pud committed:
Update tests

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308583)
* Commit: 2fd9c96dd1e38379bd8f769e77210c32e9eced98
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-2fd9c96dd1e3`

### History
* :broken_heart: [Build #308539](https://buildkite.com/elastic/kibana-pull-request/builds/308539) failed 17d657e04d4be5b245ab3b424dd06101b8396aee
* :broken_heart: [Build #308516](https://buildkite.com/elastic/kibana-pull-request/builds/308516) failed e37ae9b81e0f4b094b0714fe0e98bb15b6268651
* :broken_heart: [Build #308509](https://buildkite.com/elastic/kibana-pull-request/builds/308509) failed e3c24b42ef97982396c52f81b9892c281e58b22b
* :green_heart: [Build #308484](https://buildkite.com/elastic/kibana-pull-request/builds/308484) succeeded f5d1f8fa16a0ae54f8510e913e99591d7b085cc8
* :yellow_heart: [Build #308212](https://buildkite.com/elastic/kibana-pull-request/builds/308212) was flaky 8ce3913429ec94767b6f8b6f943b2cb7b4f7eb1c

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308583","number":308583,"commit":"2fd9c96dd1e38379bd8f769e77210c32e9eced98"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308539","number":308539,"commit":"17d657e04d4be5b245ab3b424dd06101b8396aee"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308516","number":308516,"commit":"e37ae9b81e0f4b094b0714fe0e98bb15b6268651"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308509","number":308509,"commit":"e3c24b42ef97982396c52f81b9892c281e58b22b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308484","number":308484,"commit":"f5d1f8fa16a0ae54f8510e913e99591d7b085cc8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308212","number":308212,"commit":"8ce3913429ec94767b6f8b6f943b2cb7b4f7eb1c"}],"number":308583}
buildkite-pr-comment-->

e40pud committed:
Review feedback

* Move attack discovery specific logic into a separate folder
* Rename `referencedAlerts` into `internallyManagedAlerts`

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Add tests coverage for the `groupAttackDiscoveryAlerts`

michaelolo24 commented on x-pack/platform/plugins/shared/cases/public/components/system_actions/cases/cases_params.tsx:
```diff
@@ -274,6 +310,22 @@ export const CasesParamsFieldsComponent: React.FunctionComponent<
+    return templateSelectorComponent;
+  }
+
+  return (
```
Why break apart all of the sections into memoized sub sections? If so, more of a nit here, but I would create separate components (i.e. `<GroupByComponent {...} />`) rather than wrapping the different sections in `useMemo(() => { ... })`.

e40pud replied:
It allows me easily to return only the `templateSelectorComponent` section in case of attack discovery rule type and makes code cleaner IMO. Otherwise, I would need to hide all the unnecessary parts within the `return` with `{!isAttackDiscoveryRuleType && (...)}`.

Regarding separate components that can be done, but I was thinking that it is better to do in a separate PR to keep the intend of this one scoped only to the needed updates, taking into account the time pressure 😄

andrew-goldstein reviewed (approved):
Thank you @e40pud for this feature! 🙏 
Also thanks for paring on the review
LGTM 🚀

viduni94 reviewed (approved):
Obs AI Assistant change LGTM

e40pud committed:
Fix CI

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309257)
* Commit: 396a87c3aadb93dc1948d2db8fc67dc643162cbf
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-396a87c3aadb`

### History
* :broken_heart: [Build #309211](https://buildkite.com/elastic/kibana-pull-request/builds/309211) failed 396a87c3aadb93dc1948d2db8fc67dc643162cbf
* :green_heart: [Build #308931](https://buildkite.com/elastic/kibana-pull-request/builds/308931) succeeded d99b55c7ba1b0d60c2924bcc69168faf5f489fef
* :broken_heart: [Build #308844](https://buildkite.com/elastic/kibana-pull-request/builds/308844) failed 36ed6b38c1efbb742378cb6cf68782cc0962b528
* :green_heart: [Build #308583](https://buildkite.com/elastic/kibana-pull-request/builds/308583) succeeded 2fd9c96dd1e38379bd8f769e77210c32e9eced98
* :broken_heart: [Build #308539](https://buildkite.com/elastic/kibana-pull-request/builds/308539) failed 17d657e04d4be5b245ab3b424dd06101b8396aee

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309257","number":309257,"commit":"396a87c3aadb93dc1948d2db8fc67dc643162cbf"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309211","number":309211,"commit":"396a87c3aadb93dc1948d2db8fc67dc643162cbf"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308931","number":308931,"commit":"d99b55c7ba1b0d60c2924bcc69168faf5f489fef"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308844","number":308844,"commit":"36ed6b38c1efbb742378cb6cf68782cc0962b528"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308583","number":308583,"commit":"2fd9c96dd1e38379bd8f769e77210c32e9eced98"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308539","number":308539,"commit":"17d657e04d4be5b245ab3b424dd06101b8396aee"}],"number":309257}
buildkite-pr-comment-->

cnasikas reviewed (commented):
Code review with the assumption that the Security team tested meticulously the PR. The general architecture looks good to me. I left some comments. Could you please also add integration tests that verifies the end to end integration? This can be done on a follow up PR.

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud committed:
Review feedback: Disable templates and show explaning tooltip

e40pud committed:
Review feedback: Use `ruleTypeId` instead of `alertTypeId`

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311951)
* Commit: 26c1c9709ea44c03dfbf5922288340adf16c5e7e

### Failed CI Steps
* [SLO Plugin @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/311951#0197a152-e4b9-4bdf-a9fb-3310a6302967)
* [Synthetics @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/311951#0197a152-e4ba-48cd-b0db-0f8a52c58fdc)
* [Uptime @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/311951#0197a152-e4ba-4dad-b289-9bb0b8dcb185)
* [Exploratory View @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/311951#0197a152-e4bb-4882-a8a2-d6b346ff438a)
* [UX Plugin @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/311951#0197a152-e4bc-43cf-8020-11a1163b4090)
* [UX Plugin @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/311951#0197a173-17c0-4a62-9f94-45015ad0fcce)

### History
* :broken_heart: [Build #311916](https://buildkite.com/elastic/kibana-pull-request/builds/311916) failed 699a5eabb2ed8d9984a99ea0c6da90f05a528a64
* :green_heart: [Build #309257](https://buildkite.com/elastic/kibana-pull-request/builds/309257) succeeded 396a87c3aadb93dc1948d2db8fc67dc643162cbf
* :broken_heart: [Build #309211](https://buildkite.com/elastic/kibana-pull-request/builds/309211) failed 396a87c3aadb93dc1948d2db8fc67dc643162cbf
* :green_heart: [Build #308931](https://buildkite.com/elastic/kibana-pull-request/builds/308931) succeeded d99b55c7ba1b0d60c2924bcc69168faf5f489fef
* :broken_heart: [Build #308844](https://buildkite.com/elastic/kibana-pull-request/builds/308844) failed 36ed6b38c1efbb742378cb6cf68782cc0962b528

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311951","number":311951,"commit":"26c1c9709ea44c03dfbf5922288340adf16c5e7e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311916","number":311916,"commit":"699a5eabb2ed8d9984a99ea0c6da90f05a528a64"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309257","number":309257,"commit":"396a87c3aadb93dc1948d2db8fc67dc643162cbf"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309211","number":309211,"commit":"396a87c3aadb93dc1948d2db8fc67dc643162cbf"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308931","number":308931,"commit":"d99b55c7ba1b0d60c2924bcc69168faf5f489fef"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308844","number":308844,"commit":"36ed6b38c1efbb742378cb6cf68782cc0962b528"}],"number":311951}
buildkite-pr-comment-->

e40pud commented on x-pack/platform/plugins/shared/alerting/server/task_runner/action_scheduler/schedulers/system_action_scheduler.ts:
```diff
@@ -144,6 +144,7 @@ export class SystemActionScheduler<
           name: this.context.rule.name,
           consumer: this.context.rule.consumer,
           producer: this.context.ruleType.producer,
+          alertTypeId: this.context.rule.alertTypeId,
```
Done in https://github.com/elastic/kibana/pull/222827/commits/e91c34d60b1cf546934a1edce9c5091265ea275a

e40pud committed:
Fix CI

e40pud committed:
Update x-pack/platform/plugins/shared/cases/server/connectors/cases/schema.ts

Co-authored-by: Christos Nasikas <xristosnasikas@gmail.com>

e40pud committed:
Fix CI

e40pud committed:
Update x-pack/platform/plugins/shared/cases/server/connectors/cases/schema.ts

Co-authored-by: Christos Nasikas <xristosnasikas@gmail.com>

e40pud committed:
Fix CI

e40pud committed:
Review feedback: Use `MAX_ALERTS_PER_CASE`

e40pud commented on x-pack/platform/plugins/shared/cases/server/connectors/cases/schema.ts:
```diff
@@ -65,6 +65,13 @@ const TimeWindowSchema = schema.string({
 });
 
+export const CasesGroupedAlertsSchema = schema.object({
+  alerts: schema.arrayOf(AlertSchema),
```
Done in https://github.com/elastic/kibana/pull/222827/commits/cc0db813170586baec9b583baea4494408e41096

e40pud committed:
Review feedback: Comments limit

e40pud commented on x-pack/platform/plugins/shared/cases/server/connectors/cases/schema.ts:
```diff
@@ -65,6 +65,13 @@ const TimeWindowSchema = schema.string({
 
+export const CasesGroupedAlertsSchema = schema.object({
+  alerts: schema.arrayOf(AlertSchema),
+  comments: schema.maybe(schema.arrayOf(schema.string())),
```
Done in https://github.com/elastic/kibana/pull/222827/commits/13c2297fc62866b77f336e3194289cbd26826e67

@cnasikas just curious why do we want to limit comments to `MAX_DOCS_PER_PAGE / 2` and not to `MAX_DOCS_PER_PAGE`?

cnasikas replied:
There is a limit in cases where the count of total comments + total user actions should be less than `MAX_DOCS_PER_PAGE`. I arbitrarily decided that 5K (half of `MAX_DOCS_PER_PAGE`) is a good limit to handle this scenario :slightly_smiling_face:.

e40pud committed:
Review feedback: Limit title length

e40pud commented on x-pack/platform/plugins/shared/cases/server/connectors/cases/schema.ts:
```diff
@@ -65,6 +65,13 @@ const TimeWindowSchema = schema.string({
+  alerts: schema.arrayOf(AlertSchema),
+  comments: schema.maybe(schema.arrayOf(schema.string())),
+  grouping: schema.recordOf(schema.string(), schema.any()),
+  title: schema.maybe(schema.string()),
```
Done in https://github.com/elastic/kibana/pull/222827/commits/0d6acc915690660490eccbf22f821c8d6923bb3a

e40pud committed:
Review feedback: Namings

e40pud commented on x-pack/platform/plugins/shared/cases/server/connectors/cases/cases_connector_executor.ts:
```diff
@@ -1135,23 +1142,34 @@ export class CasesConnectorExecutor {
-            alertId: alerts.map((alert) => alert._id),
-            index: alerts.map((alert) => alert._index),
+      ({ theCase, alerts, comments }) => {
+        const commentAttachment: UserCommentAttachmentPayload[] =
```
Done in https://github.com/elastic/kibana/pull/222827/commits/d92df4e3bac440ec5607c46c68fef48a22ee9a87

e40pud committed:
Review feedback: No `useMemo` for primitive values

e40pud committed:
Review feedback: Fall back to the general flow if AD grouping throws an error

e40pud commented on x-pack/platform/plugins/shared/cases/server/connectors/cases/index.ts:
```diff
@@ -98,6 +101,17 @@ export const getCasesConnectorAdapter = ({
+      let internallyManagedAlerts = false;
+      let groupedAlerts: CasesGroupedAlerts[] | null = null;
+      if (rule.alertTypeId === ATTACK_DISCOVERY_SCHEDULES_ALERT_TYPE_ID) {
+        groupedAlerts = groupAttackDiscoveryAlerts(caseAlerts);
```
Done in https://github.com/elastic/kibana/pull/222827/commits/4b4d36d46a09739d4f47d150e17e07d4561b444e

e40pud committed:
Review feedback: AD alerts grouping curcuit breakers

e40pud commented on x-pack/platform/plugins/shared/cases/server/connectors/cases/attack_discovery/index.ts:
```diff
@@ -0,0 +1,63 @@
+     * Each attack discovery alert references a list of SIEM alerts that led to the attack.
+     * These SIEM alerts will be added to the case.
+     */
+    return {
```
Done in https://github.com/elastic/kibana/pull/222827/commits/bd22e1f6bb7e4e37fc7912865b64623477c57052

cnasikas commented on x-pack/platform/plugins/shared/cases/server/connectors/cases/schema.ts:
```diff
@@ -65,6 +65,13 @@ const TimeWindowSchema = schema.string({
+export const CasesGroupedAlertsSchema = schema.object({
+  alerts: schema.arrayOf(AlertSchema),
+  comments: schema.maybe(schema.arrayOf(schema.string())),
+  grouping: schema.recordOf(schema.string(), schema.any()),
```
This is a bit different. It is before the grouping. An empty `groupingBy` means not grouping, all alerts to the same case. The `grouping` is a bit different. It means for each record we create, one case. We could allow pass `0` if you like. It is not a breaking change because the `groupedAlerts` is a new attribute, optional, and the `groupingBy` is not taken into account when the `groupedAlerts` is provided. Did I miss something?

e40pud committed:
Fix CI

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

e40pud commented:
@michaelolo24 @cnasikas I addressed review feedback and plan to address the following in a separate PR:

1. Move Cases action components into separate files
  * https://github.com/elastic/kibana/pull/222827#discussion_r2150081638
2. Tests coverage:
  * https://github.com/elastic/kibana/pull/222827#discussion_r2156665084
  * https://github.com/elastic/kibana/pull/222827#discussion_r2156754830
  * https://github.com/elastic/kibana/pull/222827#discussion_r2156773711
  * https://github.com/elastic/kibana/pull/222827#discussion_r2156774999

e40pud committed:
Review feedback: Remove breaking changes

e40pud committed:
Basic connector adapter tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312126)
* Commit: bb6527629822feb970d9cc5bc531486693f00737
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-bb6527629822`

### History
* :broken_heart: [Build #312053](https://buildkite.com/elastic/kibana-pull-request/builds/312053) failed bd22e1f6bb7e4e37fc7912865b64623477c57052
* :broken_heart: [Build #311951](https://buildkite.com/elastic/kibana-pull-request/builds/311951) failed 26c1c9709ea44c03dfbf5922288340adf16c5e7e
* :broken_heart: [Build #311916](https://buildkite.com/elastic/kibana-pull-request/builds/311916) failed 699a5eabb2ed8d9984a99ea0c6da90f05a528a64
* :green_heart: [Build #309257](https://buildkite.com/elastic/kibana-pull-request/builds/309257) succeeded 396a87c3aadb93dc1948d2db8fc67dc643162cbf
* :broken_heart: [Build #309211](https://buildkite.com/elastic/kibana-pull-request/builds/309211) failed 396a87c3aadb93dc1948d2db8fc67dc643162cbf

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312126","number":312126,"commit":"bb6527629822feb970d9cc5bc531486693f00737"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312053","number":312053,"commit":"bd22e1f6bb7e4e37fc7912865b64623477c57052"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311951","number":311951,"commit":"26c1c9709ea44c03dfbf5922288340adf16c5e7e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311916","number":311916,"commit":"699a5eabb2ed8d9984a99ea0c6da90f05a528a64"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309257","number":309257,"commit":"396a87c3aadb93dc1948d2db8fc67dc643162cbf"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309211","number":309211,"commit":"396a87c3aadb93dc1948d2db8fc67dc643162cbf"}],"number":312126}
buildkite-pr-comment-->

michaelolo24 reviewed (approved):
Nice work and thank you for making the changes! 👍🏾

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/cases-support

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312193)
* Commit: 1eb52d2dbf51e89fd7f21cb3c9f6fc70cbd8bde4
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222827-1eb52d2dbf51`

### History
* :green_heart: [Build #312126](https://buildkite.com/elastic/kibana-pull-request/builds/312126) succeeded bb6527629822feb970d9cc5bc531486693f00737
* :broken_heart: [Build #312053](https://buildkite.com/elastic/kibana-pull-request/builds/312053) failed bd22e1f6bb7e4e37fc7912865b64623477c57052
* :broken_heart: [Build #311951](https://buildkite.com/elastic/kibana-pull-request/builds/311951) failed 26c1c9709ea44c03dfbf5922288340adf16c5e7e
* :broken_heart: [Build #311916](https://buildkite.com/elastic/kibana-pull-request/builds/311916) failed 699a5eabb2ed8d9984a99ea0c6da90f05a528a64
* :green_heart: [Build #309257](https://buildkite.com/elastic/kibana-pull-request/builds/309257) succeeded 396a87c3aadb93dc1948d2db8fc67dc643162cbf

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312193","number":312193,"commit":"1eb52d2dbf51e89fd7f21cb3c9f6fc70cbd8bde4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312126","number":312126,"commit":"bb6527629822feb970d9cc5bc531486693f00737"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312053","number":312053,"commit":"bd22e1f6bb7e4e37fc7912865b64623477c57052"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311951","number":311951,"commit":"26c1c9709ea44c03dfbf5922288340adf16c5e7e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311916","number":311916,"commit":"699a5eabb2ed8d9984a99ea0c6da90f05a528a64"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309257","number":309257,"commit":"396a87c3aadb93dc1948d2db8fc67dc643162cbf"}],"number":312193}
buildkite-pr-comment-->

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/225137">](https://github.com/elastic/kibana/pull/225137)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
